src/database/migration/v2.js

```diff
@@ -188,6 +132,61 @@ const dataMigrations = [
       });
     },
   },
+  {
+    version: '2.3.6',
+    migrate: database => {
+      // Find all the transactions with no item batches, if any
+      const transactionBatches = database
+        .objects('TransactionBatch')
+        .filtered('itemBatch == $0', null)
+        .snapshot();
+      // Do nothing if there aren't any.
+      if (!(transactionBatches.length > 0)) return;
+      // For each of these item batchless-transaction batches, create a new
+      // 0 quantity item batch
+      transactionBatches.forEach(transactionBatch => {
+        if (!transactionBatch) return;
```

Comment:
this is probably excessive

Reply:
yeah .. especially with snapshotting, but realm lazy-but-live results are sometimes unpredictable heh